common/bsonutil/bsonutil.go

```diff
@@ -21,6 +21,7 @@ import (
 )
 
 var ErrNoSuchField = errors.New("no such field")
+var ErrNoSuchString = errors.New("field present, but is not a string")
```

Comment:
It might be worth allowing this to be formatted with the actual value that was expected to be a string.

e.g.
`field present, but it is not a string: 1234`.